crates/http/src/routes.rs

```diff
@@ -38,11 +38,29 @@ impl Router {
     /// If there are multiple possible components registered for the same route or
     /// wildcard, this returns the last entry in the component map.
     pub(crate) fn route(&self, p: &str) -> Result<&str> {
-        self.routes
-            .iter()
-            .rfind(|(rp, _)| rp.matches(p))
-            .map(|(_, c)| c.as_ref())
-            .with_context(|| format!("Cannot match route for path {}", p))
+        let matches = self.routes.iter().filter(|(rp, _)| rp.matches(p));
```

Comment:
This is probably a micro-optimisation but YOLO

If I understand the usage pattern here, this reiterates the _full_ list of routes on every request, whereas the previous version iterated only until it found a match.

Would be be worth sorting the list once, as we build it?  Then this function could go back to being "return the first match."

Reply:
cc @lann because we talked about the re-use of this logic for `lhc`. Is pre-sorting lexicographically to stateful for where `lhc` is going and how it might implement routing?

Reply:
I think this should still be lazy; `matches` is an iterator.

Reply:
That said, I'm not opposed to sorting while building either.

Reply:
Ah, yes, good point.  I amend my comment to "we iterate the full list unless we hit an exact match" (which will indeed often happen and get us out early).  Thanks!